src/utils/apub_name.rs

```diff
@@ -0,0 +1,96 @@
+use cfg_if::cfg_if;
+use std::sync::LazyLock;
+
+pub fn create_user_apub_name(actor_id: &str) -> String {
+  create_apub_name::<'@'>(actor_id)
+}
+
+pub fn create_community_apub_name(actor_id: &str) -> String {
+  create_apub_name::<'!'>(actor_id)
+}
+
+fn format_apub_name<const PREFIX: char>(instance: &str, name: &str) -> String {
+  format!("{PREFIX}{name}@{instance}")
+}
+
+fn create_apub_name<const PREFIX: char>(actor_id: &str) -> String {
+  const REGEX_STR: &str =
+    r"https?:\/\/((?:(?:[^.]\.)*[^.]+\.[^:]+)|localhost)(?::\d{1,5})?\/[uc]\/([^?#]+)";
```

Comment:
Not a perfect regex for URL parsing, but I anticipate actor IDs to consistently be the same format, allowing us to take some shortcuts. Let me know if my assumption is wrong.

Reply:
What's the purpose of this one? The actor ids should point to the correct locations already.

Reply:
> What's the purpose of this one? The actor ids should point to the correct locations already.

The actor ID is a URL to the user/comment, e.g. `https://instance.tld/u/name`. I need to do some parsing if I want to transform it into the stylized ID, e.g. `@name@instance.tld`. It's possible for me to pass the user/community name in, but sadly only the instance id is available. If I want to get the instance domain for the stylized apub links, the only place I could reliably get it was from the `actor_id`.

Reply:
Instead of regex, you can just get the hostname from the actor_id , then use the community / user name.

`@{name}@{hostName(person.actor_id)}` 

Anywhere you want to create these stylized names, you'll likely already have `Person` or `Community`, which has everything you need (name, display name, and actor_id).

Here's an [example for person and community](https://github.com/LemmyNet/jerboa/blob/main/app/src/main/java/com/jerboa/Utils.kt#L450)

Reply:
I'm using the `url` crate for the server-side and the browser URL binding for the client-side. It greatly simplified things.